### PR TITLE
[4237] Set action and reducer for submitted email on forgot password

### LINKED ISSUE #4237: Wrong error message shown after user went through Forgot password
**Description**:
Wrong error message shown after user went through Forgot password procedure: error message does not contain name of email

**Expected behavior**:
Error message with name of email shown:
`'If there is an account associated with _**elen4@mail.csdas**_ you will receive an email with a link to reset your password.'`

**Screenshots, Video, Logs**
https://vimeo.com/673083150/84d9f9ac16

**To Reproduce**:
1. Open FE
2. Open Forgot password page
3. Enter email
4. Submit

**Actual result**:
Wrong error message shown after user went through Forgot password procedure: error message does not contain name of email

### PR DESCRIPTION
**Pull request by pablobentivengo**
**Related issue(s):**
* Fixes #4237 

**Problem:**
* Wrong error message shown after user went through Forgot password procedure: error message does not contain name of email
* Error message does not contain name of email

**In this PR:**
* Set action and reducer for submitted email on MyAccount Store
* Set email variable on forgot password succes on MyAccountForgotPassword.container
* Read email variable from store on MyAccountForgotPasswordSucces Component

### CODE REVIEW

**Comment by pablobentivengo** [comment]
Notification not yet working properly.

**Comment by aleksandrakorolova** [comment]
Please, read carefully task:
**Expected behavior:**
Error message with name of email shown:
'If there is an account associated with _**elen4@mail.csdas**_ you will receive an email with a link to reset your password.'

**Actual result:**
Error message appears 'If there is an account associated with sfdsfds@sdfdsf.df the provided address you will receive an email with a link to reset your password'
Screenshot 2022-02-15 at 08 14 41

**Review by ejnshtein — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/MyAccountForgotPasswordSuccess/MyAccountForgotPasswordSuccess.component.js [hunk 1]
```diff
@@ -11,19 +11,28 @@
...
 import PropTypes from 'prop-types';
 import { PureComponent } from 'react';
+import { connect } from 'react-redux';
 
 import { SignInStateType } from 'Type/Account.type';
 
+import { mapDispatchToProps } from '../MyAccountForgotPassword/MyAccountForgotPassword.container';
+
+/** @namespace Component/MyAccountForgotPasswordSuccess/Component/mapStateToProps */
+export const mapStateToProps = (state) => ({
```
[the remark this answers is not in the document]
Comment (pablobentivengo) [reply packages/scandipwa/src/component/MyAccountForgotPasswordSuccess/MyAccountForgotPasswordSuccess.component.js]: You are right!! I created Container and moved Redux there.

File: packages/scandipwa/src/component/MyAccountForgotPasswordSuccess/MyAccountForgotPasswordSuccess.container.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                state,
+                handleSignIn,
+                submittedEmail
+            };
+        }
+
+        render() {
+            return (
+                <MyAccountForgotPasswordSuccess
+                  { ...this.containerFunctions }
```
Comment (ejnshtein) [inline packages/scandipwa/src/component/MyAccountForgotPasswordSuccess/MyAccountForgotPasswordSuccess.container.js:54]: `this.containerFunctions` should be `undefined`, no?

**Review by ejnshtein — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/MyAccountForgotPasswordSuccess/MyAccountForgotPasswordSuccess.container.js (hunk 2 above)
Comment (pablobentivengo) [reply packages/scandipwa/src/component/MyAccountForgotPasswordSuccess/MyAccountForgotPasswordSuccess.container.js]: Yeahp!

**Pull request merged**